src/types.ts

```diff
@@ -988,3 +1012,68 @@ export interface StepOpts {
    */
   id?: string;
 }
+
+/**
+ * Simplified version of Rust style `Result`
+ *
+ * Make it easier to wrap functions with some kind of result.
+ * e.g. API calls
+ */
+export type Result<T, E = undefined> =
+  | { ok: true; value: T }
+  | { ok: false; error: E | undefined };
+
+export const Ok = <T>(data: T): Result<T, never> => {
+  return { ok: true, value: data };
+};
+
+export const Err = <E>(error?: E): Result<never, E> => {
+  return { ok: false, error };
+};
+
+/**
+ * Format of data send from the executor to the SDK
+ */
+export const fnDataSchema = z.object({
+  event: z.object({}).passthrough(),
+  events: z.array(z.object({}).passthrough()).default([]),
```

Comment:
Should the default of `events` be an empty array? Do we not always expect it to be populated?

Reply:
Noting - as a follow-up, it'd be great to not send the event twice in the payload, i.e. `event` and `events[0]`. I think we talked about this before and discussed that the executor can look at the SDK version / feature set and decide what payload shape to send. Doesn't have to happen now 🙂